immuneML/reports/encoding_reports/PositionalMotifFrequencies.py

```diff
@@ -38,41 +41,63 @@ def _generate(self):
         PathBuilder.build(self.result_path)
 
         motifs = self.dataset.encoded_data.feature_names
-        plotting_data = self._get_plotting_data(motifs)
+        stacked_bar, hist_dict = self._get_plotting_data(motifs)
 
```

Comment:
there is one additional type of output that we would want here, which is to have the raw data exported so that the user can recreate the plots if they want, or so that the raw results are exported just in case there was an error during plotting. 
The stacked_bar is a dataframe so can be exported to csv directly, for hist_dict you may need to find another way to represent the data to export it (ideally looking something like csv, if this is a nested dictionary it could also be in json format)

Those additonal outputs can be added to ReportResult.output_tables (.csv files) or output_text (other files)

See for example the report SignificantKmerPositions whihc has the method _write_results_table for this purpose